analyzer/src/main/kotlin/managers/MavenDependencyHandler.kt

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (C) 2021 Bosch.IO GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.analyzer.managers
+
+import org.apache.maven.project.MavenProject
+import org.apache.maven.project.ProjectBuildingException
+
+import org.eclipse.aether.graph.DependencyNode
+
+import org.ossreviewtoolkit.analyzer.managers.utils.DependencyHandler
+import org.ossreviewtoolkit.analyzer.managers.utils.MavenSupport
+import org.ossreviewtoolkit.analyzer.managers.utils.identifier
+import org.ossreviewtoolkit.model.OrtIssue
+import org.ossreviewtoolkit.model.Package
+import org.ossreviewtoolkit.model.PackageLinkage
+import org.ossreviewtoolkit.model.createAndLogIssue
+import org.ossreviewtoolkit.utils.collectMessagesAsString
+import org.ossreviewtoolkit.utils.log
+import org.ossreviewtoolkit.utils.showStackTrace
+
+/**
+ * A specialized [DependencyHandler] implementation for the dependency model of Maven.
+ */
+class MavenDependencyHandler(
+    /** The name of the associated package manager. */
+    val managerName: String,
+
+    /** The helper object to invoke Maven-related functionality. */
+    val mvn: MavenSupport,
+
+    /**
+     * A map with information about the local projects in the current Maven build. Dependencies pointing to projects
+     * sometimes need to be treated in a special way.
+     */
+    val localProjects: Map<String, MavenProject>,
+
+    /**
+     * A flag whether [SBT compatibility mode][Maven.enableSbtMode] is enabled.
+     */
+    val sbtMode: Boolean
+) : DependencyHandler<DependencyNode> {
+    override fun identifierFor(dependency: DependencyNode): String {
+        val id = dependency.identifier()
+        val typePrefix = if (isLocalProject(id)) managerName else "Maven"
```

Comment:
Does this distinction make sense? Won't it be "Maven" in any case in the end?

Reply:
The current Sbt implementation uses this class, too. In this case, the manager name is _Sbt_.

Reply:
Shouldn't then the condition be considering `sbtMode` instead of `isLocalProject`?

Reply:
The original behavior for Maven and Sbt regarding the package type (which is not changed by this PR) is as follows:
* External dependencies are always assigned the type _Maven_.
* For local projects, the type is set to the name of the package manager.

For Maven, there is indeed no difference, as the type is always _Maven_. Sbt, however, will produce packages of type _Maven_ and _Sbt_, the latter for local projects only.

I am not sure about the background, but I assume, as the current implementation of Sbt is based on generated POM files that are processed by Maven, it makes somewhat sense to classify the packages as dependencies resolved by Maven.

Reply:
Ah, thanks for reminding me. The background is that the `type` of `Identifier`s for `Package`s should always by the type of the package (in the PURLs sense, if you want), i.e. Maven, Sbt and Gradle all consume Maven packages / artifacts. But the `type` of `Identifier`s for `Project`s should be named after the actual package manager in use, i.e. Maven, Sbt or Gradle etc.

So the code is indeed correct. Just a small nit: I'd name the variable just `type` instead of `typePrefix`, as there's actually nothing being added to the type besides this "prefix".

Reply:
Renamed variable to `type`.